### ISSUE TITLE
Error running benchmark script: Expected B.stride(-1) == 1 to be true, but got false

### ISSUE THREAD

**Issue by abacaj** [state-spaces/mamba#36]
I get this error running `benchmark` script:

```
File ~/personal/transformer-experiments/env/lib/python3.10/site-packages/mamba_ssm/ops/selective_scan_interface.py:37, in SelectiveScanFn.forward(ctx, u, delta, A, B, C, D, z, delta_bias, delta_softplus, return_last_state)
     35     C = rearrange(C, "b dstate l -> b 1 dstate l")
     36     ctx.squeeze_C = True
---> 37 out, x, *rest = selective_scan_cuda.fwd(u, delta, A, B, C, D, z, delta_bias, delta_softplus)
     38 ctx.delta_softplus = delta_softplus
     39 ctx.has_z = z is not None

RuntimeError: Expected B.stride(-1) == 1 to be true, but got false.  (Could this error message be improved?  If so, please report an enhancement request to PyTorch.)
```

Env:
```
Linux, cuda 12.1

torch==2.1.0
triton==2.1.0
mamba-ssm==1.0.1
causal-conv1d==1.0.1
```

**Comment by geronimi73** [comment]
had the same when the prompt is too short (for whatever reason)
```
python3 benchmark_generation_mamba_simple.py --prompt "hello"
```
=> fails
```
python3 benchmark_generation_mamba_simple.py --prompt "hello this is long enough"
```
=> **works**

**Comment by robflynnyh** [comment]
had this same problem when trying to train from scratch

**Comment by tridao** [comment]
This is fixed in v1.1.0

**Comment by mas-38** [comment]
Recently I used Mamba v1.0.1 that i download a long time ago before, with the input of a ViT style tensor, when i set the batchsize=1, the same Error occured. I try the method like that 

mamba_simple.py
```
 dt = self.dt_proj.weight @ dt.t()
                dt = rearrange(dt, "d (b l) -> b d l", l=seqlen)
### ---add here to made tensor really contiguous at dim=-1
                if batch == 1:
                    B = B.clone().reshape(batch, self.d_state, seqlen)
                    C = C.clone().reshape(batch, self.d_state, seqlen)
             
                else:
                    B = rearrange(B, "(b l) dstate -> b dstate l", b = batch,l=seqlen).contiguous()
                    C = rearrange(C, "(b l) dstate -> b dstate l", b = batch,l=seqlen).contiguous()
```

and in selective_scan_interface.py, 
```
class SelectiveScanFn,    
@staticmethod
    def backward(ctx, dout, *args):
        if not ctx.has_z:
            u, delta, A, B, C, D, delta_bias, x = ctx.saved_tensors
            z = None
            out = None
        else:
            u, delta, A, B, C, D, z, delta_bias, x, out = ctx.saved_tensors
        if dout.stride(-1) != 1:
###---modify here
            #dout = dout.contiguous()
            b,n,c = dout.shape
            dout = dout.reshape(b, n * c).unsqueeze(-1).contiguous().view(b, n, c)
```

for the same reason, then you can running with batchsize = 1 without the Error. Hope it can help for old version users.
